3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -107,11 +107,17 @@ function add_member() {
   "
 
   # TODO: replace this with a call to `replset_addr` from common.sh, once it returns host names.
-  local replset_addr
-  replset_addr="${MONGODB_REPLICA_NAME}/$(find_endpoints | paste -s -d,)"
+  local endpoints
+  endpoints="$(find_endpoints | paste -s -d,)"
+
+  if [ -z "${endpoints}" ]; then
+    info "ERROR: couldn't join replica set!"
+    info "CAUSE: couldn't find hosts that are belongs to the replica set."
+    return 1
```

Comment:
@rhcarvalho When we decided to move retry logic into JavaScript we were wrong actually: there is non-zero possibility that potential member can't connect to replica set at all because primary isn't ready yet.